python/oneflow/nn/optimizer/sgd.py

```diff
@@ -72,9 +72,12 @@ def __init__(
         self._default_options["weight_decay"] = weight_decay
         if isinstance(parameters, collections.abc.Iterator):
             self.param_groups.append(ParamGroup(parameters, self._default_options))
-        else:
-            for param in parameters:
-                self.param_groups.append(ParamGroup(param, self._default_options))
+        elif isinstance(parameters, collections.abc.Iterable):
+            if isinstance(parameters[0], dict):
+                for param in parameters:
+                    self.param_groups.append(ParamGroup(param, self._default_options))
+            else:
+                self.param_groups.append(ParamGroup(parameters, self._default_options))
```

Comment:
```suggestion
                self.param_groups.append(ParamGroup(parameters.__iter__(), self._default_options))
```

这样写怎么样呢，ParamGroup里统一只接收迭代器或dict

Reply:
> 这样写怎么样呢，ParamGroup里统一只接收迭代器或dict

这里要把整个List[Tensor]传进去，假如只用__iter__的话self._parameter没办法一下子接受整个list，就必须进行append操作会导致变成非叶张量

Reply:
我这里写了个例子试了试是没问题的，Parameter应该都是浅拷贝，不应该会出现变非叶子结点的情况